server/pulp/server/async/scheduler.py

```diff
@@ -121,30 +113,38 @@ def check_celery_processes(self):
         present. If there are zero of either, log at the error level that Pulp will not operate
         correctly.
         """
-        msg = _('Checking if pulp_workers, pulp_celerybeat, or pulp_resource_manager '
-                'processes are missing for more than %d seconds') % self.CELERY_TIMEOUT_SECONDS
+        msg = _('Checking if pulp_workers, pulp_celerybeat, or pulp_resource_manager processes '
+                'are missing for more than %d seconds') % constants.CELERY_TIMEOUT_SECONDS
         _logger.debug(msg)
         now = ensure_tz(datetime.utcnow())
-        oldest_heartbeat_time = now - timedelta(seconds=self.CELERY_TIMEOUT_SECONDS)
+        oldest_heartbeat_time = now - timedelta(seconds=constants.CELERY_TIMEOUT_SECONDS)
         worker_list = Worker.objects.all()
         worker_count = 0
         resource_manager_count = 0
         scheduler_count = 0
+
         for worker in worker_list:
             if worker.last_heartbeat < oldest_heartbeat_time:
                 msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
                 _logger.error(msg)
-                _delete_worker(worker.name)
-            elif worker.name.startswith(SCHEDULER_WORKER_NAME):
+
+                if not worker.name.startswith(constants.SCHEDULER_WORKER_NAME):
```

Comment:
This would be slightly more tidy if you drop the ``not`` and swap lines 132 and 134. That makes a nice flow for readability:

If this is the special case involving the scheduler:
  Do the scheduler-specific thing.
Otherwise:
  Do the normal thing.

Note that the two scheduler-related lines are together this way.